ebu_tt_live/node/denester.py

```diff
@@ -0,0 +1,332 @@
+from .base import AbstractCombinedNode
+from ebu_tt_live.documents.ebutt3 import EBUTT3Document
+from pyxb.binding.basis import NonElementContent, ElementContent
+from ebu_tt_live.bindings import div_type, p_type, span_type, ebuttdt, style_type
+from ebu_tt_live.bindings._ebuttm import divMetadata_type
+import copy
+import re
+from datetime import timedelta
+
+class DenesterNode(AbstractCombinedNode):
+    """
+    This class contains methods for unnesting elements of an EBUTT3Document that cannot be nested in EBUTTD
+    """
+
+    _expects = EBUTT3Document
+    _provides = EBUTT3Document
+
+    def __init__(self, node_id, consumer_carriage=None, producer_carriage=None):
+        super(DenesterNode, self).__init__(
+            node_id=node_id,
+            producer_carriage=producer_carriage,
+            consumer_carriage=consumer_carriage
+        )
+
+    def process_document(self, document, **kwargs):
+        divs = document.binding.body.div
+        unnested_divs = []
+        dataset = {}
+        if document.binding.head.styling is not None:
+            dataset["styles"] = document.binding.head.styling.style
+        else:
+            dataset["styles"] = []
+        dataset["document"] = document.binding
+        for div in divs:
+            unnested_divs.extend(DenesterNode.recurse(div,dataset))
+        unnested_divs = DenesterNode.combine_divs(unnested_divs)
+        unnested_divs = DenesterNode.check_p_regions(unnested_divs)
+        document.binding.body.div = unnested_divs
+        return document
```

Comment:
Testing the Denester, no output documents are generated at all here. As far as I can see, returning the document is not enough but instead `self.producer_carriage.emit_data` has to be called to forward the processed document.

Reply:
Agreed, I see that I have some changes made on the BBC fork that include a fix for that, so I will push an update to this pull request.